lib/repl.js

```diff
@@ -282,17 +282,24 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
+    var out;
     internalUtil.decorateErrorStack(e);
-    if (e instanceof SyntaxError && e.stack) {
+
+    if (!((e instanceof Error) && typeof e.stack === 'string')) {
+      out = e;
```

Comment:
@cjihrig I think [this](https://github.com/cjihrig/node-1/blob/98286455ef4e1913fae176eb42a468b60b6db5a0/lib/repl.js#L257-L260) might be the missing puzzle piece. ;)